File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.py
# Time Complexity: O(N)
# Space Complexity: O(1)

class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        N = len(nums)
        curr_sum = nums[0]
        max_sum = curr_sum
        for i in range(1, N):
            curr_sum = curr_sum + nums[i] if nums[i] > nums[i-1] else nums[i]
            max_sum = max(max_sum, curr_sum)
        return max_sum

File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.rs
// Time Complexity: O(N)
// Space Complexity: O(1)

use std::cmp::max;
impl Solution {
    pub fn max_ascending_sum(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut curr_sum = nums[0];
        let mut max_sum = curr_sum;

        for i in (1..n){
            if nums[i] > nums[i-1] {
                curr_sum += nums[i]
            }
            else {
                curr_sum = nums[i]
            }
            max_sum = max(max_sum, curr_sum);
        }
        max_sum
    }
}
